lib/charms/tls_certificates_interface/v2/tls_certificates.py

```diff
@@ -1237,9 +1242,9 @@ def certificate_issued_for_csr(self, app_name: str, csr: str) -> bool:
             bool: True/False depending on whether a certificate has been issued for the given CSR.
         """
         issued_certificates_per_csr = self.get_issued_certificates()[app_name]
-        for request, cert in issued_certificates_per_csr.items():
-            if request == csr:
-                return csr_matches_certificate(csr, cert)
+        for request_pair in issued_certificates_per_csr:
+            if "csr" in request_pair:
```

Comment:
Before we were checking if the value stored was equal to `csr` before returning the results of `csr_matches_certificate()`. Now, we're just returning `csr_matches_certificate()` for the first entry that has a `csr` key in it and disregarding the match. 

This looks like a change in behaviour to me, but I can't tell if it is intentional or why we no longer need to do this check.

Reply:
Before, we were returning a dictionary of CSR's for the specific application. This is problematic because multiple CSR's could be coming from the same application. This would not error out, but overwrite the previous CSR when generating issued certificates.

Now, we are returning an array of dicts with keys {csr, certificate}. This allows us to return multiple pairs of csrs for one application. This also requires us to loop over the array to find the correct {csr, certificate} pair. The array could be empty, or have multiple csr's.

Now that I type this out, I see your point. Previously, there was an assumption that only one CSR could be issued per application, so no unit test checks if the second CSR was the one that was being checked. I've fixed it.